src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -112,166 +157,272 @@ class boss_thaddius : public CreatureScript
+                            BeginResetEncounter();
+                        break;
+                    case ACTION_FEUGEN_AGGRO:
+                    case ACTION_STALAGG_AGGRO:
+                        if (_phase == PHASE_RESETTING)
+                            return BeginResetEncounter();
+                        if (_phase != PHASE_NOT_ENGAGED)
+                            return;
+                        _phase = PHASE_PETS;
+                        events.SetPhase(PHASE_PETS);
+
+                        shockingEligibility = true;
+                        
+                        if (!instance->CheckRequiredBosses(BOSS_THADDIUS))
+                            return BeginResetEncounter();
+                        instance->SetBossState(BOSS_THADDIUS, IN_PROGRESS);
+
+                        me->setActive(true);
+                        if (Creature* stalagg = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_STALAGG)))
+                            stalagg->setActive(true);
+                        if (Creature* feugen = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FEUGEN)))
+                            feugen->setActive(true);
+                        break;
+                    case ACTION_FEUGEN_DIED:
+                        if (Creature* feugen = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FEUGEN)))
+                            feugen->AI()->DoAction(ACTION_FEUGEN_REVIVING_FX);
+                        feugenAlive = false;
+                        if (stalaggAlive)
+                            events.ScheduleEvent(EVENT_REVIVE_FEUGEN, 5 * IN_MILLISECONDS, 0, PHASE_PETS);
+                        else
+                            Transition();
+
+                        break;
+                    case ACTION_STALAGG_DIED:
+                        if (Creature* stalagg = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_STALAGG)))
+                            stalagg->AI()->DoAction(ACTION_STALAGG_REVIVING_FX);
+                        stalaggAlive = false;
+                        if (feugenAlive)
+                            events.ScheduleEvent(EVENT_REVIVE_STALAGG, 5 * IN_MILLISECONDS, 0, PHASE_PETS);
+                        else
+                            Transition();
+
+                        break;
+
+                    case ACTION_POLARITY_CROSSED:
+                        shockingEligibility = false;
+                        break;
+                    default:
+                        break;
+                }
             }
 
-            if (!checkFeugenAlive && !checkStalaggAlive)
+            uint32 GetData(uint32 id) const override
             {
-                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_STUNNED);
-                // REACT_AGGRESSIVE only reset when he takes damage.
-                DoZoneInCombat();
+                return (uint32)(id == DATA_POLARITY_CROSSED && shockingEligibility);
```

Comment:
Why the conversion? The C++ standard says:
§4.7/4 from the C++ Standard says (Integral Conversion)

If the source type is bool, the value false is converted to zero and the value true is converted to one.

So there seems to be an implicit conversion